tests/WeakMapTest.php

```diff
@@ -259,6 +261,34 @@ public function testNoInternalCycle() : void
         self::assertNull($rV->get());
     }
 
+    public function testHousekeepingOnGcRunSurvival() : void
+    {
+        $weakMap = new WeakMap();
+
+        $vkPairs = [];
+        for ($i = 100; $i > 0; $i--) {
+            for ($j = 100; $j > 0; $j--) {
+                $k = new stdClass;
+                $v = new stdClass;
+                $weakMap[$k] = $v;
+                $vkPairs[] = [$k, $v];
+            }
+
+            gc_collect_cycles();
+            gc_collect_cycles();
+            gc_collect_cycles();
+        }
+
+        foreach ($vkPairs as [$k, $v]) {
+            if ($weakMap[$k] !== $v) {
+                self::assertSame($v, $weakMap[$k]);
```

Comment:
I don't get this test (impossible given the condition above), why not `assertSame()` directly?

Reply:
just to lower the assertions count reported by PHPUnit, also it is a little faster